docs/specifications/tests/DNSSEC-TP/dnssec02.md

```diff
@@ -3,125 +3,202 @@
-
-1. Create an empty set, "DS records".
-
-2. If the *Test Type* is "undelegated, then:
-   1. Add *Undelegated DS* set to *DS records*.
-
-3. Else, do:
-   1. Create a DS query with DO flag set for the name of the
-      *Child Zone* (*Test Type* is "normal").
-   2. Retrieve all name server IP addresses for the parent zone of
-      *Child Zone* using [Method1] ("Parent NS IP").
-   3. For each IP address in *Parent NS IP* do:
-      1. Send the DS query over UDP and collect the response.
-      2. If there is no DNS response, then output *[NO_RESPONSE]*.
-      3. Else, if the RCODE is not NOERROR, then output
-         *[UNEXPECTED_RESPONSE_DS]*.
-      4. Else, go to next IP address if there is no DS in the
-         response.
-      5. Else, extract all DS records from the DNS response and
-         add them to *DS Records*.
-
-4. If *DS Records* is empty, exit this test case.
-
-5. Create a DNSKEY query with DO flag set for the apex of the
-   *Child Zone*.
-
-6. Retrieve all name server IP addresses for the
-   *Child Zone* using [Method4] and [Method5] ("NS IP").
-
-7. For each name server IP address in *NS IP* do:
-   1. Send the DNSKEY query over UDP.
-   2. If no DNS response is returned, then output *[NO_RESPONSE]*
-      and go to next name server IP.
-   3. If the DNS response does not contain an DNSKEY RRset,
-      then output *[NO_RESPONSE_DNSKEY]* and go to next name server
-      IP.
-   4. Extract the DNSKEY RRset ("DNSKEY RRs").
-   5. If the DNS response does not contain any RRSIG records covering
-      the DNSKEY RRset, then output *[NO_RRSIG_DNSKEY]*.
-   5. Extract the RRSIG records covering the DNSKEY RRset, possibly
-      none ("DNSKEY RRSIG").
-   6. For each DS in *DS Records*, do:
-      1. Find the equivalent DNSKEY in *DNSKEY RRs* by key ID (key tag).
-      2. If matching DNSKEY is not found, output *[NO_MATCHING_DNSKEY]*.
-      3. If the DS values (algorithm and digest) do not match the
-         DNSKEY then output *[BROKEN_DS]*.
-      4. If bit 7 of the DNSKEY flags field has the value of 0 (nil),
-         then output *[DNSKEY_NOT_ZONE_SIGN]* and go to next DS.
-      5. If bit 15 of the DNSKEY flags field has the value of 0 (nil),
-         then output *[DNSKEY_KSK_NOT_SEP]*.
-      6. Find the equivalent RRSIG in *DNSKEY RRSIG* by key ID (key tag).
-      7. If matching RRSIG is not found, output *[NO_MATCHING_RRSIG]*.
-      8. If the RRSIG values (algorithm and signature) do not match
-         the DNSKEY then output *[BROKEN_RRSIG]*.
-
-8. If no message, besides possibly *[NO_RESPONSE]* or 
-   *[DNSKEY_KSK_NOT_SEP]*, has been outputted, then output 
-   *[DS_MATCHES]*.
+  (only if *Test Type* is [undelegated]).
+
+
+## Summary
+
+* If not both DS records and DNSKEY records are found, then further
+inverstigation will not be done and no messages will be outputted.
+* No messages will be outputted due errors in the responses from the parent name
```

Comment:
maybe a missing word: "due errors" -> "due to errors" ?

Reply:
Fixed.